pkg/catalogsourceconfig/succeeded.go

```diff
@@ -37,7 +43,24 @@ func (r *succeededReconciler) Reconcile(ctx context.Context, in *marketplace.Cat
 	// No change is being made, so return the CatalogSourceConfig object as is.
 	out = in
 
-	r.log.Info("No action taken, the object has already been reconciled")
+	msg := "No action taken, the object has already been reconciled"
 
-	return out, nil, nil
+	// Check if the CatalogSource exists
+	catalogSourceGet := new(CatalogSourceBuilder).WithTypeMeta().CatalogSource()
+	key := client.ObjectKey{
+		Name:      in.Name,
+		Namespace: in.Spec.TargetNamespace,
+	}
+
+	if err := r.client.Get(context.TODO(), key, catalogSourceGet); err != nil {
+		// The CatalogSource has been deleted. Drop the existing Status field so
+		// that reconciliation can start anew.
+		out.Status = marketplace.CatalogSourceConfigStatus{}
+		nextPhase = phase.GetNext(phase.Configuring)
+		msg = "CatalogSource has been deleted, scheduling for configuring"
```

Comment:
Do we care about the distinction between deleted and not found?

Reply:
I don't think so as the implication is the same. The resulting configuring shouldn't hurt either.